src/Spryker/Zed/Category/Business/Updater/CategoryUrlUpdater.php

```diff
@@ -98,11 +98,13 @@ protected function executeUpdateCategoryUrlTransaction(CategoryTransfer $categor
      */
     public function updateCategoryNodeUrls(CategoryTransfer $categoryTransfer): void
     {
+        $idCategoryNode = $categoryTransfer->getCategoryNodeOrFail()->getIdCategoryNode();
```

Comment:
Before, the result was not nullable, with this it now is `int|null`
I wonder if we should go extra safe and actually keep the fallback in place for max BC?

```suggestion
        $idCategoryNode = $categoryTransfer->getCategoryNodeOrFail()->getIdCategoryNode();
        if (!$idCategoryNode) {
            $idCategoryNode = $categoryTransfer->getIdCategoryOrFail();
        }
```
This way at least one must be present/set, and if none are the existing exception would still be raised.

Reply:
Are you able to adjust the PR accordingly?

Reply:
Feel free to just take over the PR and adjust it if you want. I just applied your suggest here.